### PR TITLE
[WPF] Respect TreeView column Expands flag

### PR DESCRIPTION
**Pull request by daserge**
This will automatically expand the columns marked with Expands flag when their content changes.
Shift cell views column index and global cell indexes following the removed TreeView column.
^ This also fixes ArgumentOutOfRangeException in TreeViewBackend->GetCellBounds

### CODE REVIEW

File: Xwt.WPF/Xwt.WPFBackend/ExTreeViewItem.cs [hunk 1]
```diff
@@ -226,6 +226,11 @@ private void OnLoaded (object sender, RoutedEventArgs routedEventArgs)
 			}
 		}
 
+		public void ExpandColumnWidth (int columnIndex, double toWidth)
```
Comment (sevoku) [inline Xwt.WPF/Xwt.WPFBackend/ExTreeViewItem.cs]:
```suggestion
		public void SetColumnWidth (int columnIndex, double toWidth)
```
and does this need to be `public`?

File: Xwt.WPF/Xwt.WPFBackend/TreeViewBackend.cs [hunk 2]
```diff
@@ -234,6 +235,59 @@ public void UnselectAll ()
...
+						if (cell_bounds.Width > maxColumnViewWidth)
+							maxColumnViewWidth = cell_bounds.Width;
+					}
+
+					totalColumnWidth += maxColumnViewWidth;
+				}
+
+				treeViewItem.ExpandColumnWidth (changedColumnIndex, totalColumnWidth);
+			} catch (Exception ex) {
+				Console.WriteLine ($"[TreeViewBackend_NodeChanged] Exception: {ex}");
```
Comment (sevoku) [inline Xwt.WPF/Xwt.WPFBackend/TreeViewBackend.cs]: Is there a reason to swallow the exception here?

**Comment by daserge** [comment]
`GlobalCellIndex` is for sequential index of a CellView.

File: Xwt/Xwt/ITreeDataSource.cs [hunk 3]
```diff
@@ -68,6 +72,8 @@ public TreeNodeEventArgs (TreePosition node, int childIndex)
 			get;
 			private set;
 		}
+
+		public int ChangedCellIndex { get; private set; }
```
[the remark this answers is not in the document]
Comment (daserge) [reply Xwt/Xwt/ITreeDataSource.cs]: Actually this is the field index here and not the column: https://github.com/mono/xwt/blob/af2540cb0b6a1c1e82c4f7ec8949565f6677e631/Xwt/Xwt/TreeNavigator.cs#L121
Maybe I should rename this to `FieldIndex` for clarity?

**Comment by chkn** [comment]
@daserge @sevoku What's the status of this? We have a designer issue that would probably be resolved by this patch.

**Comment by daserge** [comment]
@chkn I've encountered more issues with column width calculations which I didn't find a solution to yet.
The Designer issue has a workaround currently so I consider this as a lower priority task.
Let me reopen when its ready.

**Pull request closed without merging**